test/e2e/network_test.go

```diff
@@ -558,22 +595,53 @@ func (tc *testContext) deleteJob(name string) error {
 func (tc *testContext) waitUntilJobSucceeds(name string) error {
 	var job *batchv1.Job
 	var err error
+	var labelSelector string
 	for i := 0; i < retryCount; i++ {
 		job, err = tc.client.K8s.BatchV1().Jobs(tc.workloadNamespace).Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
 			return err
 		}
+		labelSelector = "job-name=" + job.Name
 		if job.Status.Succeeded > 0 {
+			tc.savePodLogsAndEvents(labelSelector)
 			return nil
 		}
 		if job.Status.Failed > 0 {
-			events, _ := tc.getPodEvents(name)
-			return errors.Errorf("job %v failed: %v", job, events)
+			tc.savePodLogsAndEvents(labelSelector)
+			return errors.Errorf("job %v failed.", job)
 		}
 		time.Sleep(retryInterval)
 	}
-	events, _ := tc.getPodEvents(name)
-	return errors.Errorf("job %v timed out: %v", job, events)
+	tc.savePodLogsAndEvents(labelSelector)
+	return errors.Errorf("job %v timed out.", job)
+}
+
+// savePodLogsAndEvents prints the logs and events associated with the
+// label selector of a given job or deployment to the ARTIFACT_DIR
+func (tc *testContext) savePodLogsAndEvents(labelSelector string) {
+	events, err := tc.getPodEvents(labelSelector)
+	if err != nil {
+		log.Printf("Unable to get events associated with pod label: %s", labelSelector)
+	} else {
+		log.Printf("Events for pod %s: %v", labelSelector, events)
+	}
+	logs, err := tc.getLogs(labelSelector)
+	if err != nil {
+		log.Printf("Unable to get logs associated with pod: %s", labelSelector)
+	} else {
```

Comment:
You can add a `return` here and get rid of the `else`. It will make the code easier to read.